### ISSUE TITLE
Warn about useless variable overwriting?

### ISSUE THREAD

**Issue by Ocramius** [kalessil/phpinspectionsea#472]
Here's a bit of code that is functionally correct, yet semantically incorrect and hard to read:

```php
$output = str_replace('foo', 'bar', $input);
$output = trim($output);
$output = htmlentities($output);
$output = '<p>' . $output . '</p>';
```

Why is this incorrect? The reason is simple: the semantic meaning of the `$output` variable changes.

While this may turn into a very annoying inspection, it is appropriate to require the author of the code to give each step of this piece of code a different variable with a different naming that reflects the meaning of it.

**Comment by kalessil** [comment]
What about suggesting `$output = htmlentities(trim(str_replace('foo', 'bar', $input)))` instead?

**Comment by Ocramius** [comment]
Also OK, but most of the time it involves non-functional operations

**Comment by kalessil** [comment]
Kind of getting deeply nested array values?

**Comment by Ocramius** [comment]
Even just $foo = $bar->baz(); if (cond) {  $foo = $bar->taz($foo);  }

**Comment by kalessil** [comment]
Ok, I see now: I'll check which patterns we can extract from this code style.

**Comment by kalessil** [comment]
Pilot version implemented, new patterns will come once people accommodated with initial version and requests more patterns to be detected.
